### PR TITLE
feat(glog): add glog in planning and control modules

### PR DESCRIPTION
**Pull request by TakaHoribe**
## Description

Add a glog component and enable it in planning and control modules.

This enables making a stack-trace message before the process dies as below:

```

[component_container_mt-28] this should raise an exception: *** Aborted at 1692707767 (unix time) try "date -d @1692707767" if you are using GNU date ***
[component_container_mt-28] PC: @                0x0 (unknown)
[component_container_mt-28] *** SIGSEGV (@0xfffffffffffffffc) received by PID 224474 (TID 0x7f259e7fc640) from PID 18446744073709551612; stack trace: ***
[component_container_mt-28]     @     0x7f25b8724046 (unknown)
[component_container_mt-28]     @     0x7f25cc242520 (unknown)
[component_container_mt-28]     @     0x7f2599c7abac behavior_path_planner::StartPlannerModule::plan()
[component_container_mt-28]     @     0x7f2599c77f43 behavior_path_planner::StartPlannerModule::run()
[component_container_mt-28]     @     0x7f2599b2ee37 behavior_path_planner::PlannerManager::run()
[component_container_mt-28]     @     0x7f2599b1d9f4 behavior_path_planner::PlannerManager::runApprovedModules()
[component_container_mt-28]     @     0x7f2599b22665 _ZZN21behavior_path_planner14PlannerManager3runERKSt10shared_ptrINS_11PlannerDataEEENKUlvE0_clEv
[component_container_mt-28]     @     0x7f2599b23189 behavior_path_planner::PlannerManager::run()
[component_container_mt-28]     @     0x7f2599b54abc behavior_path_planner::BehaviorPathPlannerNode::run()
[component_container_mt-28]     @     0x7f2599b5cfd5 rclcpp::GenericTimer<>::execute_callback()
[component_container_mt-28]     @     0x7f25cca54de1 rclcpp::Executor::execute_any_executable()
[component_container_mt-28]     @     0x7f25cca5c09a rclcpp::executors::MultiThreadedExecutor::run()
[component_container_mt-28]     @     0x7f25cc6dc253 (unknown)
[component_container_mt-28]     @     0x7f25cc294b43 (unknown)
[component_container_mt-28]     @     0x7f25cc326a00 (unknown)
[component_container_mt-28]     @                0x0 (unknown)
[ERROR] [component_container_mt-28]: process has died [pid 224474, exit code -11, cmd '/opt/ros/humble/lib/rclcpp_components/component_container_mt --ros-args -r __node:=behavior_planning_container -r __ns:=/planning/scenario_planning/lane_driving/behavior_planning -p use_sim_time:=False -p wheel_radius:=0.383 -p wheel_width:=0.235 -p wheel_base:=2.79 -p wheel_tread:=1.64 -p front_overhang:=1.0 -p rear_overhang:=1.1 -p left_overhang:=0.128 -p right_overhang:=0.128 -p vehicle_height:=2.5 -p max_steer_angle:=0.7']
```

## Related links

None
## Tests performed

run psim

## Notes for reviewers

None

## Interface changes

None
## Effects on system behavior

The process will show a stack-trace message before it dies.

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [x] The PR follows the [pull request guidelines].
- [x] The PR has been properly tested.
- [x] The PR has been reviewed by the code owners.

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [x] There are no open discussions or they are tracked via tickets.
- [x] The PR is ready for merge.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

**Review by kyoichi-sugahara — APPROVED** [review]

File: launch/tier4_control_launch/launch/control.launch.py [hunk 1]
```diff
@@ -306,6 +306,13 @@ def launch_setup(context, *args, **kwargs):
         target_container="/control/control_container",
     )
 
+    glog_component = ComposableNode(
+        package="glog_component",
+        plugin="glog_component::GlogComponent",
```
Comment (wep21) [inline launch/tier4_control_launch/launch/control.launch.py:311]: @TakaHoribe The plugin seems to be registered as `GlogComponent`, so I guess you should add namespace into component implementation or remove namespace here. Please check if this component is loaded by this launch.

File: launch/tier4_planning_launch/launch/scenario_planning/lane_driving/behavior_planning/behavior_planning.launch.py [hunk 2]
```diff
@@ -64,6 +64,13 @@ def launch_setup(context, *args, **kwargs):
     with open(LaunchConfiguration("behavior_path_planner_param_path").perform(context), "r") as f:
         behavior_path_planner_param = yaml.safe_load(f)["/**"]["ros__parameters"]
 
+    glog_component = ComposableNode(
+        package="glog_component",
+        plugin="glog_component::GlogComponent",
```
Comment (wep21) [inline launch/tier4_planning_launch/launch/scenario_planning/lane_driving/behavior_planning/behavior_planning.launch.py:69]: ditto

File: launch/tier4_planning_launch/launch/scenario_planning/lane_driving/motion_planning/motion_planning.launch.py [hunk 3]
```diff
@@ -309,6 +309,17 @@ def launch_setup(context, *args, **kwargs):
         condition=IfCondition(LaunchConfiguration("use_surround_obstacle_check")),
     )
 
+    glog_component = ComposableNode(
+        package="glog_component",
+        plugin="glog_component::GlogComponent",
```
Comment (wep21) [inline launch/tier4_planning_launch/launch/scenario_planning/lane_driving/motion_planning/motion_planning.launch.py:314]: ditto

File: launch/tier4_planning_launch/launch/scenario_planning/lane_driving/motion_planning/motion_planning.launch.py [hunk 4]
```diff
@@ -309,6 +309,17 @@ def launch_setup(context, *args, **kwargs):
...
+        name="glog_component",
+        extra_arguments=[{"use_intra_process_comms": True}],
```
Comment (wep21) [inline launch/tier4_planning_launch/launch/scenario_planning/lane_driving/motion_planning/motion_planning.launch.py:316]:
```suggestion
```

File: launch/tier4_planning_launch/launch/scenario_planning/lane_driving/behavior_planning/behavior_planning.launch.py [hunk 5]
```diff
@@ -64,6 +64,13 @@ def launch_setup(context, *args, **kwargs):
...
+        name="glog_component",
+        extra_arguments=[{"use_intra_process_comms": True}],
```
Comment (wep21) [inline launch/tier4_planning_launch/launch/scenario_planning/lane_driving/behavior_planning/behavior_planning.launch.py:71]:
```suggestion
```

File: launch/tier4_control_launch/launch/control.launch.py [hunk 6]
```diff
@@ -306,6 +306,13 @@ def launch_setup(context, *args, **kwargs):
...
+        name="glog_component",
+        extra_arguments=[{"use_intra_process_comms": True}],
```
Comment (wep21) [inline launch/tier4_control_launch/launch/control.launch.py:313]:
```suggestion
```

File: common/glog_component/include/glog_component/glog_component.hpp [hunk 7]
```diff
@@ -0,0 +1,28 @@
+// Copyright 2023 Tier IV, Inc.
```
Comment (wep21) [inline common/glog_component/include/glog_component/glog_component.hpp:1]:
```suggestion
// Copyright 2023 TIER IV, Inc.
```

File: common/glog_component/src/glog_component.cpp [hunk 8]
```diff
@@ -0,0 +1,25 @@
+// Copyright 2023 Tier IV, Inc.
```
Comment (wep21) [inline common/glog_component/src/glog_component.cpp:1]:
```suggestion
// Copyright 2023 TIER IV, Inc.
```

File: launch/tier4_control_launch/launch/control.launch.py (hunk 1 above)
Comment (TakaHoribe) [reply launch/tier4_control_launch/launch/control.launch.py]: Thank you! I forgot to push some local commits. Not it should work!

File: launch/tier4_planning_launch/launch/scenario_planning/lane_driving/behavior_planning/behavior_planning.launch.py (hunk 2 above)
Comment (TakaHoribe) [reply launch/tier4_planning_launch/launch/scenario_planning/lane_driving/behavior_planning/behavior_planning.launch.py]: done

File: launch/tier4_planning_launch/launch/scenario_planning/lane_driving/motion_planning/motion_planning.launch.py (hunk 3 above)
Comment (TakaHoribe) [reply launch/tier4_planning_launch/launch/scenario_planning/lane_driving/motion_planning/motion_planning.launch.py]: done

File: common/glog_component/src/glog_component.cpp (hunk 8 above)
Comment (TakaHoribe) [reply common/glog_component/src/glog_component.cpp]: Done

File: common/glog_component/include/glog_component/glog_component.hpp (hunk 7 above)
Comment (TakaHoribe) [reply common/glog_component/include/glog_component/glog_component.hpp]: Done

**Review by takayuki5168 — APPROVED** [review]

**Review by wep21 — APPROVED** [review]
LGTM

**Pull request merged**